### PR TITLE
Update security considerations for CSI inline ephemeral volumes

### PR DESCRIPTION
**Pull request by dobsonj**
This PR:
- Updates the milestones for Generic Ephemeral Inline Volumes (already went GA in 1.23)
- Updates the documentation for CSI inline volumes per the KEP's [Security Considerations](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/596-csi-inline-volumes#security-considerations) and [Read-only Volumes](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/596-csi-inline-volumes#read-only-volumes) sections.

KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/596-csi-inline-volumes
Enhancement: https://github.com/kubernetes/enhancements/issues/596

```release-note
Update security considerations for CSI inline ephemeral volumes
```

### CODE REVIEW

**Comment by dobsonj** [comment]
/retitle Update security considerations for CSI inline ephemeral volumes
/cc @jsafrane

**Comment by jsafrane** [comment]
/lgtm
I'd prefer second pair of eyes to approve this too.

**Comment by jsafrane** [comment]
cc @msau42 @jingxu97

**Comment by dobsonj** [comment]
Latest push just mirrors the grammatical corrections from https://github.com/kubernetes/website/pull/32667

File: book/src/ephemeral-local-volumes.md [hunk 1]
```diff
@@ -116,6 +118,17 @@ supports dynamic provisioning of volumes. No other changes are needed
...
 
+## Security Considerations
+
+CSI driver vendors that choose to support ephemeral inline volumes are responsible for secure handling of these volumes, and special consideration needs to be given to what volumeAttributes are supported by the driver. As noted above, a CSI driver is not suitable for CSI ephemeral inline volumes when volume creation requires volumeAttributes that should be restricted to an administrator. These attributes are set directly in the Pod spec, and therefore are not automatically restricted to an administrator when used as an inline volume.
+
+CSI inline volumes are only intended to be used for ephemeral storage, and driver vendors should NOT allow usage of inline volumes for persistent storage unless they also provide a third party pod admission plugin to restrict usage of these volumes.
+
+Cluster administrators who need to restrict the CSI drivers that are
+allowed to be used as inline volumes within a Pod spec may do so by:
+- removing `Ephemeral` from `volumeLifecycleModes` in the CSIDriver spec
```
Comment (xing-yang) [inline book/src/ephemeral-local-volumes.md:129]: Can you change this to be the same as what you have in the K8s website doc PR?
```
Removing `Ephemeral` from `volumeLifecycleModes` in the CSIDriver spec, which prevents the driver from being used as an inline ephemeral volume.
```

Comment (dobsonj) [reply book/src/ephemeral-local-volumes.md]: Updated, thanks!

**Pull request merged**